pkg/deploy/controller/deploymentcancellation/factory.go

```diff
@@ -0,0 +1,70 @@
+package deploymentcancellation
+
+import (
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	kutil "github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+
+	controller "github.com/openshift/origin/pkg/controller"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// DeploymentCancellationControllerFactory can create a DeploymentCancellationController
+// that cancels deployments that have the cancellation annotation set.
+type DeploymentCancellationControllerFactory struct {
+	// KubeClient is a Kubernetes client.
+	KubeClient kclient.Interface
+	// Codec is used for encoding/decoding.
+	Codec runtime.Codec
+}
+
+// Create creates a DeploymentCancellationController.
+func (factory *DeploymentCancellationControllerFactory) Create() controller.RunnableController {
+	deploymentLW := &deployutil.ListWatcherImpl{
```

Comment:
Added a TODO in the code. Currently field selectors are not supported for replication controllers upstream.